File: Problems/Binary/Missing-Number/Missing-Number.cs
using System;

// Using XOR operation - Time: O(n)
public class Solution
{
    public int MissingNumber(int[] nums)
    {
        int n = nums.Length;
        int xorVal = 0;
        for (int i = 0; i < n; i++)
            xorVal ^= nums[i] ^ (i + 1);

        return xorVal;
    }
}

class Program
{
    static void Main(string[] args)
    {
        int[] nums = { 3, 0, 1 };
        Console.WriteLine("Input: nums = " + string.Join(", ", nums));

        Solution sol = new Solution();  
        int ans = sol.MissingNumber(nums);

        Console.WriteLine("Output: " + ans);
    }
}

File: Problems/Binary/Missing-Number/Missing-Number.cpp
#include <iostream>
#include <vector>

// Using XOR operation - Time: O(n)

class Solution {
public:
    int missingNumber(std::vector<int> &nums)
    {
        int n = nums.size(); 
        int xorVal = 0;
        
        for (int i = 0; i < n; ++i)
            xorVal ^= nums[i] ^ (i + 1);

        return xorVal;
    }
};

int main()
{
    std::vector<int> nums = {3, 0, 1}; 
    
    Solution sol;
    int ans = sol.missingNumber(nums);

    std::cout << "Input: nums = [";
    for (int num : nums) 
    {
        std::cout << num << "";
        if (num != nums.back())
            std::cout << ", ";
    }
    std::cout << "]\n";

    std::cout << "Output: " << ans << std::endl;

    return 0;
}